metric_name_builder.go

```diff
@@ -81,29 +80,55 @@ var perUnitMap = map[string]string{
 	"y":  "year",
 }
 
+// MetricNameBuilder is a helper struct to build metric names.
+type MetricNameBuilder struct {
+	namespace      string
+	metricSuffixes bool
+}
+
+// NewDefaultMetricNameBuilder creates a new MetricNameBuilder with default values:
+// - namespace: ""
+// - metricSuffixes: false
+//
+// Please use NewMetricNameBuilder for custom configurations.
+func NewDefaultMetricNameBuilder() *MetricNameBuilder {
+	return NewMetricNameBuilder("", false)
+}
+
+// NewMetricNameBuilder creates a new MetricNameBuilder.
+//
+// namespace is the namespace to prefix the metric name with.
+// metricSuffixes controls the addition of metric suffixes to the metric name, e.g. unit suffixes
+// and type suffixes (e.g. "_total" for counters).
+func NewMetricNameBuilder(namespace string, metricSuffixes bool) *MetricNameBuilder {
```

Comment:
```suggestion
func NewMetricNameBuilder(...Options) *MetricNameBuilder {
```

Thinking a bit more, are we ok with variadic options when creating the helper struct at least? If not, adding new functionality will continue to be a breaking change 😕

Reply:
Why not simply adding fields to `MetricNameBuilder`?

Reply:
Yeah, in my mind, all the fields were supposed to be unexported. I've updated the PR to a point where I'm satisfied with the result, and we don't have constructors.

I have on PoC up, any thoughts? :)